com.unity.render-pipelines.high-definition/Runtime/Lighting/Shadow/ScreenSpaceShadowManagerPunctual.RenderGraph.cs

```diff
@@ -0,0 +1,161 @@
+using System;
+using UnityEngine.Experimental.Rendering;
+using UnityEngine.Experimental.Rendering.RenderGraphModule;
+
+namespace UnityEngine.Rendering.HighDefinition
+{
+    public partial class HDRenderPipeline
+    {
+        TextureHandle DenoisePunctualScreenSpaceShadow(RenderGraph renderGraph, HDCamera hdCamera,
+                                                    HDAdditionalLightData additionalLightData, in LightData lightData,
+                                                    TextureHandle depthBuffer, TextureHandle normalBuffer, TextureHandle motionVetorsBuffer,
+                                                    TextureHandle noisyBuffer, TextureHandle velocityBuffer, TextureHandle distanceBuffer)
+        {
+            // Is the history still valid?
+            float historyValidity = EvaluateHistoryValidityPointShadow(hdCamera, lightData, additionalLightData);
+
+            // Evaluate the channel mask
+            GetShadowChannelMask(lightData.screenSpaceShadowIndex, ScreenSpaceShadowType.GrayScale, ref m_ShadowChannelMask0);
+
+            // Apply the temporal denoiser
+            HDTemporalFilter temporalFilter = GetTemporalFilter();
+            HDTemporalFilter.TemporalDenoiserArrayOutputData temporalFilterResult;
+            if (additionalLightData.distanceBasedFiltering)
```

Comment:
Two branches seems completely similar apart from the shadowHistoryDistanceArray parameter. Can we merge the two branch and just do a ternary on the parameter to reduce code duplication?